cylc/flow/parsec/exceptions.py

```diff
@@ -94,10 +94,11 @@ def __str__(self):
         if self.line:
             msg += ":\n   " + self.line.strip()
         if self.lines:
-            msg += "\nContext lines:\n" + "\n".join(self.lines)
-            msg += "\t<--"
-            if self.err_type:
-                msg += ' %s' % self.err_type
+            for filename, lines in self.lines.items():
```

Comment:
I think `self.lines` is still a list in the Jinja2 case (as opposed to empy)? Maybe you forgot to push up the next commit...

Reply:
I replaced it with a dict in `cylc/flow/parsec/jinja2support.py`, it should now print both the `{% include` line and the "included" line.

Reply:
(When I first looked, there were Jinja2 functional test errors relating to `self.lines` type here; but nevermind, you've clearly fixed it one way or another).